src/Stratis.Bitcoin.Features.SmartContracts/PoA/SmartContractPoARuleRegistration.cs

```diff
@@ -25,68 +30,54 @@ public class SmartContractPoARuleRegistration : IRuleRegistration
         private readonly ISenderRetriever senderRetriever;
         private readonly IReceiptRepository receiptRepository;
         private readonly ICoinView coinView;
-        private readonly PoAConsensusRulesRegistration baseRuleRegistration;
+       // private readonly PoAConsensusRulesRegistration baseRuleRegistration;
         private readonly IEnumerable<IContractTransactionPartialValidationRule> partialTxValidationRules;
         private readonly IEnumerable<IContractTransactionFullValidationRule> fullTxValidationRules;
 
-        public SmartContractPoARuleRegistration(Network network,
-            IStateRepositoryRoot stateRepositoryRoot,
-            IContractExecutorFactory executorFactory,
-            ICallDataSerializer callDataSerializer,
-            ISenderRetriever senderRetriever,
-            IReceiptRepository receiptRepository,
-            ICoinView coinView,
-            IEnumerable<IContractTransactionPartialValidationRule> partialTxValidationRules,
-            IEnumerable<IContractTransactionFullValidationRule> fullTxValidationRules)
+        public SmartContractPoARuleRegistration()
         {
-            this.baseRuleRegistration = new PoAConsensusRulesRegistration();
-            this.network = network;
-            this.stateRepositoryRoot = stateRepositoryRoot;
-            this.executorFactory = executorFactory;
-            this.callDataSerializer = callDataSerializer;
-            this.senderRetriever = senderRetriever;
-            this.receiptRepository = receiptRepository;
-            this.coinView = coinView;
-            this.partialTxValidationRules = partialTxValidationRules;
-            this.fullTxValidationRules = fullTxValidationRules;
         }
 
-        public virtual void RegisterRules(IConsensus consensus)
+        public virtual void RegisterRules(IServiceCollection services)
```

Comment:
As I pointed above, I think this class could just be a static class that's an extension of IServiceCollection so we can just register rules using an extension like `services.RegisterSmartContractPoaRule()`

Why did we need a class and all these injected stuff?
I don't think we need this

Reply:
I did nto want to mess around with SC logic too much, this preserves the current logic and will go away in a following PR by someone with better understanding of SC rules logic.